broke-broker/src/main/java/org/dsngroup/broke/broker/storage/ServerSession.java

```diff
@@ -0,0 +1,19 @@
+package org.dsngroup.broke.broker.storage;
+
+import io.netty.handler.codec.mqtt.MqttMessage;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ServerSession {
+
+    final String clientId;
+
+    // Unacked messages store: Key: packet idenfier, value: message
+    final ConcurrentHashMap<String, MqttMessage> unackedMessages;
+
```

Comment:
Can't get this, what's this for?

Reply:
This is for the future implementation of message delivery semantics(Mqtt QoS). For example, before the QoS 1 publisher receives its corresponding PUBACK, the PUBLISH message will be stored in this unackedMessages hash map.

Reply:
OK, but to get the waiting ack message, is `BlockingQueue` with associate frame better?
i.e.
```java
private class UnAckedMessageFrame {
    String packetIdentifier;
    MqttMessage message;
}
```

Reply:
Oh, my mistake, ignore this, thanks!

Reply:
I was considering that the unacked messages may need to be deleted once the ack is received. That's why I chose hash map for efficient search & deletion. 
But if considers the amount of messages, an array-like message queue with an ack recording message set or map might be better. 
I'll consider this carefully again.

Reply:
Yes, this depends on how you implement `QoS1` mechanism.